docs/architecture/ingest.md

```diff
@@ -0,0 +1,87 @@
+# Ingest
+
+This is a component that is responsible for ingesting and processing data.
+
+It exposes a GRPC API for Osmosis node clients to send data to be ingested.
+
+See [this ADR](https://www.notion.so/osmosiszone/ADR-006-Streaming-GRPC-SQS-Ingest-4e3b2ff7d23e43e2a1f3c43adc3c26bc) for details about the design.
+
+The node pushes only the pool data that is changed within a block.
+
+As a result, it is possible to see the following sequence of events:
+- Height X: All Osmosis pools are pushed
+- Height X+1: Only the pools that have changed within that height are pushed
+
+## Workers
+
+### Pricing
+
+The pricing worker is responsible for pre-computing prices for all assets and the default quote denom (USDC).
+
+It is triggerred by the ingester asyncronously when the GRPC push is received.
+
+It looks at the unique denoms constructed from the pool data and computes the prices for each
+asset and the default quote denom.
+
+Once complete, it calls a hook to notify the subscribed listeners that the prices have been updated.
+
+#### Pricing Listeners
+
+- Healthcheck: The healthcheck listener is responsible for updating the healthcheck status based on the last time the prices were updated. If the prices are not updated within a certain time period, the healthcheck status will be updated to unhealthy.
+An example of a listener is the healtcheck component. Once it is notified of the price update, it
+- Pool Liquidity Pricing Worker: This worker is responsible for updating the pool liquidity capitalization
+based on the prices that are computed by the pricing worker.
+
+### Pool Liquidity Pricing
+
+The pool liquidity pricing worker is responsible for computing liquidity capitalization (configured in USDC by default).
+
+There are 2 kinds of liquidities capitalizations that we compute:
+1. Token liquidity capitalization across all pools
+  * Value of all pool liquidity of a token across all pools
+2. Pool liqudity capitalization
+  * Value of all pool balances in USDC
+
+To compute the above, we need to know:
+1. Prices for all tokens
+2. Liquidity for all tokens in all pools
+
+There are 2 possble cases:
```

Comment:
Be nice we can have example of the data in case 1 and 2

Reply:
Could you please suggest something specific?

I'm not quite sure how to construct them to be useful

Reply:
Nevermind this. I've already added a suggestion in the beginning of this doc.